### ISSUE TITLE
Resolve Gradle warnings

### ISSUE THREAD

**Issue by EwoutH** [minvws/nl-covid19-notification-app-android#53]
### Describe the bug, issue or concern
Resolve the following Gradle issues

**Warnings:**
 - [x] API 'BaseVariant.getApplicationIdTextResource' is obsolete
```
WARNING: API 'BaseVariant.getApplicationIdTextResource' is obsolete and has been replaced with 'VariantProperties.applicationId'.
It will be removed in version 5.0 of the Android Gradle plugin.
```
 - [x] permission was tagged at AndroidManifest.xml:16 to remove other declarations but no other declaration present
```
/home/runner/work/nl-covid19-notification-app-android/nl-covid19-notification-app-android/app/src/debug/AndroidManifest.xml:16:5-18:31 Warning:
	uses-permission#android.permission.READ_EXTERNAL_STORAGE was tagged at AndroidManifest.xml:16 to remove other declarations but no other declaration present
/home/runner/work/nl-covid19-notification-app-android/nl-covid19-notification-app-android/app/src/debug/AndroidManifest.xml:19:5-21:31 Warning:
	uses-permission#android.permission.WRITE_EXTERNAL_STORAGE was tagged at AndroidManifest.xml:19 to remove other declarations but no other declaration present
/home/runner/work/nl-covid19-notification-app-android/nl-covid19-notification-app-android/app/src/debug/AndroidManifest.xml:22:5-24:31 Warning:
	uses-permission#android.permission.READ_PHONE_STATE was tagged at AndroidManifest.xml:22 to remove other declarations but no other declaration present
```

**Depreciations:**

- [x] kotlin.text.Appendable /* = java.lang.Appendable */' is deprecated. Use appendLine instead. Note that the new method always appends the line feed character '\n' regardless of the system line separator.
- [x] 'setter for systemUiVisibility: Int' is deprecated. Deprecated in Java
- [x] 'getter for systemUiVisibility: Int' is deprecated. Deprecated in Java
- [x] 'SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN: Int' is deprecated. Deprecated in Java
- [x] 'SYSTEM_UI_FLAG_LAYOUT_STABLE: Int' is deprecated. Deprecated in Java
- [x] 'EXTRA_TOKEN: String' is deprecated. Deprecated in Java
- [x] 'quarantineDays: Int' is deprecated. Removed in v3 endpoint. Kept for compatibility will be removed in a future release
- [x] 'ExposureSummary' is deprecated. Deprecated in Java
- [x] 'ExposureConfiguration' is deprecated. Deprecated in Java

Here's the [full build log](https://github.com/EwoutH/nl-covid19-notification-app-android/runs/1702149636)

### To Reproduce
Build with Gradle 6.8 with `--warning-mode=all`, for example `./gradlew app:assemDevDebug testDevDebug testDebug app:lintDevDebug --warning-mode=all`

### Expected behavior
Gradle `--warning-mode=fail` should pass, see #54.

### Governance
- [x] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [x] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [x] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

**Comment by ehoogend** [comment]
Hi @EwoutH , Thanks for reporting the issue.

Some of the warnings are from libraries we use and will not drop. For example the first warning (API 'BaseVariant.getApplicationIdTextResource' is obsolete) is actually an issue from the navigation library's safe args gradle plugin (see https://issuetracker.google.com/issues/159542337) that should be fixed when using Android Gradle Plugin 4.2 and up)

The warnings related to system UI visibility and flags have been resolved by moving to the new WindowCompat api ([see this commit](https://github.com/minvws/nl-covid19-notification-app-android/commit/213ed711f82d03f1806bfc9f7fc74befab99bf7d)).

We're working on resolving several of the remaining warnings (as well as new ones that have popped up) in the current sprint.

**Comment by lpappalettera** [comment]
All warnings above are fixed and building with `--warning-mode=fail` does succeed. There is a new warning about removing `jcenter`, but i don't think we need this issue for that.
